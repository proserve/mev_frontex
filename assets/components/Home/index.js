import React from 'react';

import './home.scss';
import MEVTable from '../MEVTable';
import Cards from '../Cards';
import { CARDS } from '../../common/utils';

export default class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      transactionsData: [],
    };
  }

  componentWillMount() {
    this.setState({ isLoading: true });
    fetch('/data/transactions.json').then((resp) => {
      resp.json().then((data) => {
        this.setState({ isLoading: false, transactionsData: data });
      });
    }).catch(() => {
      this.setState({ isLoading: false });
      alert('an error occur while fetching data');
    });
  }

  render() {
    const { isLoading, transactionsData } = this.state;
    return (
      <div className="container">
        <div className="breadcrumbs">
          <span className="left">customer list</span>
          <span> / customer details</span>
        </div>
        <div className="details">
          <span className="title">account details</span>
          <Cards cards={CARDS} />
        </div>
        {isLoading ? <h1 style={{ textAlign: 'center' }}>Loading...</h1>
          : <MEVTable transactions={transactionsData} /> }
      </div>
    );
  }
}
