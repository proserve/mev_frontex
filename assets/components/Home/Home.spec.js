import React from 'react';
import 'isomorphic-fetch';
import { shallow } from 'enzyme';
import Home from './index';

describe('Home', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Home />);
    expect(component).toMatchSnapshot();
  });
});
