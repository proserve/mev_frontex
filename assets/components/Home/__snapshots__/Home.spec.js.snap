// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly in "debug" mode 1`] = `
<div
  className="container"
>
  <div
    className="breadcrumbs"
  >
    <span
      className="left"
    >
      customer list
    </span>
    <span>
       / customer details
    </span>
  </div>
  <div
    className="details"
  >
    <span
      className="title"
    >
      account details
    </span>
    <Cards
      cards={
        Array [
          Object {
            "color": "yellow",
            "name": "Trading Limit",
            "price": "AED 193,655,500.00",
          },
          Object {
            "color": "green",
            "name": "Security Deposit",
            "price": "AED 50,000,000.00",
          },
          Object {
            "color": "red",
            "name": "Trading Limit",
            "price": "AED 5,344,500.00",
          },
        ]
      }
    />
  </div>
  <h1
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Loading...
  </h1>
</div>
`;
